### PR TITLE
GC Counters for .NET 5

### PR DESCRIPTION
**Pull request by sywhang**
This is the feature design PR for the GC counters to be added in .NET 5, as suggested in https://github.com/dotnet/runtime/pull/38851#issuecomment-657540335.

cc @tommcdon @noahfalk @Maoni0 @benmwatson @mjsabby for discussions.

### CODE REVIEW

File: docs/design/features/gc-counters-net5.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+* GC Heap Fragmentation %
+* Gen 0/1/2 Pause Time
+* Gen 0/1/2 Time Between Each GC
+
+## Solution
+
+The proposal is to add the following GC counters:
+
+* GC Heap Fragmentation %
+* Gen 0/1/2 Pause Time
```
Comment (jkotas) [inline docs/design/features/gc-counters-net5.md:29]: How is the pause time defined for Gen2 GC?

**Comment by sywhang** [comment]
@benmwatson, I had a conversation with @noahfalk about these counters and thought about the proposal myself again for quite a bit... and concluded that the diagnostics value added by the GC Pause Time counters do not merit the potential confusion by everyone else. 

Specifically, the pause time for gen 2 GC can fluctuate a huge amount depending on whether it was FGC or BGC. If the user understands the pause time characteristics for BGC or FGC, that wouldn't be a problem, but if we add this to the default list of counters emitted by the runtime, it means everyone who uses counters will be able to see these values, and I am not sure if that is obvious to most .NET developers. 

Moreover it is not really clear what scenario is enabled by adding these counters that isn't possible with the current list of GC counters. (other than the obvious one of being able to compute pause time without taking a ETW/EventPipe trace). It seems like the value added isn't clear enough for us to risk confusing a large portion of developers using counters on .NET 5.

A potential solution for you would be to implement these counters using a custom counter, instead of adding this to the runtime, since the API for computing these metrics are already public. If you need help implementing those counters, please let me know and I would be happy to help. 

For the fragmentation counter, I think there is less of a confusion along what it means and I think it can be added safely unless @Maoni0 has any other thoughts?

File: docs/design/features/gc-counters-net5.md (hunk 1 above)
Comment (sywhang) [reply docs/design/features/gc-counters-net5.md:29]: It was defined to be the latest Gen 2 GC's pause time... whether that was FGC or BGC. I spoke with Noah a bit and came to the conclusion that it's not worth adding the pause counters (discussed below)

**Comment by benmwatson** [comment]
Fair enough. We do have the custom counters right now and we can continue to use the.
________________________________

**Comment by sywhang** [comment]
Thanks @benmwatson. I will close this PR as discussed.

**Pull request closed without merging**